zerver/views/registration.py

```diff
@@ -170,10 +170,13 @@ def accounts_register(
     if realm_creation:
         role = UserProfile.ROLE_REALM_OWNER
 
-    try:
-        validators.validate_email(email)
-    except ValidationError:
-        return render(request, "zerver/invalid_email.html", context={"invalid_email": True})
+    if len(email) == 0:
+        assert realm_creation
```

Comment:
Because the email is an empty string we need to skip the email validation. I put in an assertion for realm creation being true, but I'm not sure if it would make sense to try and get the `is_demo_organization` data from the request here as well.

Reply:
We should definitely also assert `is_demo_organization` here as well.

Reply:
Continued this discussion [here](https://github.com/zulip/zulip/pull/22666/files#r949544828) with updated code.